include/nbl/video/utilities/SIntendedSubmitInfo.h

```diff
@@ -48,22 +49,20 @@ struct SIntendedSubmitInfo final : core::Uncopyable
         //
         inline ISemaphore::SWaitInfo getFutureScratchSemaphore() const {return {scratchSemaphore.semaphore,scratchSemaphore.value+1};}
         
-        // Returns the scratch to use if valid, nullptr otherwise
-        inline const IQueue::SSubmitInfo::SCommandBufferInfo* getCommandBufferForRecording() const
+        // Returns the command buffer to use for recording if valid, nullptr otherwise
+        inline const IQueue::SSubmitInfo::SCommandBufferInfo* getCurrentRecordingCommandBufferInfo() const
         {
-            if (!queue || scratchCommandBuffers.empty() || !scratchSemaphore.semaphore)
+            if (scratchCommandBuffers.empty())
                 return nullptr;
+
```

Comment:
the queue checks etc are meant to be here, see my latest refactor https://github.com/Devsh-Graphics-Programming/Nabla/commit/c1d5f3418cd6ede043a9b38749a990a81932fe6a